library/alloc/src/collections/btree/map.rs

```diff
@@ -3248,5 +3238,114 @@ impl<'a, K: Ord, V, A: Allocator + Clone> CursorMut<'a, K, V, A> {
     }
 }
 
+impl<'a, K: Ord, V, A: Allocator + Clone> CursorMut<'a, K, V, A> {
```

Comment:
As mentioned on the tracking issue, I would much rather use the entry API with key mutation instead of coming up with an entirely different API mirrored on the cursor. Methods like these could potentially just return `VacantEntry` or `OccupiedEntry` to do insertion and mutation.

Reply:
> Methods like these

I'm confused, which functions are you referring to?

I think it would be less confusing for both the entry API and the Cursor API to both have the core methods like mutating/changing entries rather than the Cursor API returning `VacantEntry`/`OccupiedEntry` which seems like it would make things quite messy as it adds an extra level of indirection.

Reply:
There is a downside to methods that consume `self` and return a new type: it makes it much harder to write a type which wraps a `CursorMut`. The methods on the wrapper type would take `&mut self` but you can't easily take ownership of the cursor if it is in a struct field.

Reply:
That's fair, although we could always just mutably borrow in these cases; the only reason why entries take values by self is because they actually contain values that are consumed. Since the lifetime is still tracked in these cases, we don't have to worry about any aliasing issues.